dirs, overlord, state: add simple file based state backend

mvo5 opened the pull request.

zyga commented:
I'd s/StateFsBackend/FsStateBackend/g because it feels more natural to read. Otherwise +1

niemeyer commented on overlord/state/fsbackend.go:
```diff
@@ -0,0 +1,36 @@
+}
+
+func NewStateFsBackend(path string) *StateFsBackend {
+	return &StateFsBackend{backFn: path}
```
Some suggestions:

* `StateFsBackend` => `fileBackend` (private)
* `NewStateFsBackend` => `NewFileBackend`, returning `StateBackend` (the interface)
* `backFn` => `path`

niemeyer commented on overlord/overlord.go:
```diff
@@ -41,9 +47,25 @@ type Overlord struct {
-	// TODO: read it or create a fresh one and learn about the system
-	// current state
-	s := state.New(nil)
+	backend := state.NewStateFsBackend(dirs.SnapStateFile)
```
I suggest moving all of that new logic into a new local private function:

`func loadState() (*state.State, error)`

niemeyer commented on overlord/overlord.go:
```diff
@@ -41,9 +47,25 @@ type Overlord struct {
+		}
+
+	} else {
+		s = state.New(backend)
```
This might be done earlier as a shortcut:

`if !FileExists { return state.New(..), nil }`

niemeyer commented:
Some trivials, and LGTM!

mvo5 commented:
Thanks for the great feedback (I really like fileBackend much better than the original name!), I updated the branch and added some missing tests. Please let me know if there is anything (or if I have overlooked anything).

niemeyer commented:
Very nice and simple, thanks @mvo5!
